GeomIO/pFIOServerBounds.F90

```diff
@@ -56,16 +56,22 @@ function get_file_shape(this) result(file_shape)
       file_shape =this%file_shape
    end function get_file_shape
 
-   function new_pFIOServerBounds(grid, field_shape, time_index, rc) result(server_bounds)
+   function new_pFIOServerBounds(grid, in_field_shape, time_index, rc) result(server_bounds)
       type(ESMF_Grid), intent(in) :: grid
-      integer, intent(in) :: field_shape(:)
+      integer, intent(in) :: in_field_shape(:)
       integer, intent(in), optional :: time_index
       integer, intent(out), optional :: rc
       type(pFIOServerBounds) :: server_bounds
 
+      integer, allocatable :: field_shape(:)
       integer :: status, tile_count, n_dims, tm, global_dim(3)
       integer :: i1, in, j1, jn, tile, extra_file_dim, file_dims, new_grid_dims
 
+      field_shape = in_field_shape
+      if (size(in_field_shape)==1) then ! vertical-only field
```

Comment:
I do wonder how fragile this logic is.    What if someone has a surface loc stream for their geom?

Reply:
I think you need to access the metadata that is added during Field construction.  That, or you need to retrieve the geom and do analysis to determine if this 1 dimension is vertical or not.

Reply:
It might be alright in this case since the first argument to the routine is a grid. We will need a separate routine for a locstream.